crates/provider/src/shared/helpers.rs

```diff
@@ -0,0 +1,75 @@
+use std::collections::HashMap;
+
+use configuration::shared::constants::VALID_REGEX;
+use regex::{Captures, Regex};
+
+pub(crate) fn apply_replacements(text: &str, replacements: &HashMap<&str, &str>) -> String {
```

Comment:
We can also support `tera` as templating lang (as we do in v1 with nunjucks) but for the moment is a little bit overkill.

Reply:
I don't understand the purpose of this function, is it to update a running k8s resource ?

Reply:
No, the idea behind this `fn` is to allow us to have manifest (`yaml` files) that we need to inject values (e.g namespace or others) when we want to deploy them, so we can just past the _template_ and inject the values. This will also be handy to manage some info that is generated by the orchestrator and needs to be injected later in the command or arguments.